dora/core/client/fs/src/main/java/alluxio/client/block/stream/DataWriter.java

```diff
@@ -56,35 +54,18 @@ private Factory() {} // prevent instantiation
     public static DataWriter create(FileSystemContext context, long blockId, long blockSize,
         WorkerNetAddress address, OutStreamOptions options) throws IOException {
       AlluxioConfiguration alluxioConf = context.getClusterConf();
-      boolean workerIsLocal = CommonUtils.isLocalHost(address, alluxioConf);
       boolean nettyTransEnabled =
           alluxioConf.getBoolean(PropertyKey.USER_NETTY_DATA_TRANSMISSION_ENABLED);
-
-      if (workerIsLocal) {
-        LOG.debug("Creating worker process local output stream for block {} @ {}",
-            blockId, address);
-        return BlockWorkerDataWriter.create(context, blockId, blockSize, options);
-      }
-      LOG.debug("Doesn't create worker process local output stream for block {} @ {} "
-          + "(data locates in local worker: {}, client locates in local worker process: {})",
-          blockId, address, workerIsLocal, context.hasProcessLocalWorker());
-
-      boolean domainSocketSupported = NettyUtils.isDomainSocketSupported(address);
```

Comment:
i actually prefer to remove the domain socket feature in a separate PR, if we never deprecated that feature in one PR. I figure we need something like short-circuit/domain-socket in the near future and we could just revert (part of) that PR.

Reply:
With Dora and netty, if we do want short circuit read/write in the future, I'm thinking about implementing it via a transfer of file descriptors over a unix domain socket. Code that is related to the detection of local worker and short circuit io, as well as high level IO api like WorkerClient should not be impacted, but low level transport related code will need to be written from scratch and little can be reused.